code/Equipment/Tools/JetpackTool.cs

```diff
@@ -0,0 +1,141 @@
+using Sandbox;
+using Sandbox.Utility;
+
+namespace Grubs.Equipment.Tools;
+
+public sealed class JetpackTool : Tool
+{
+	private float _jetpackDir { get; set; }
+
+	[Property] private float MaxJetFuel { get; set; } = 10f;
+
+	private float _currentJetFuel { get; set; }
+
+	public override void OnHolster()
+	{
+		if ( Equipment.Grub != null )
+			Equipment.Grub.Animator.IsOnJetpack = false;
+
+		FBFlame.Enabled = false;
+		UDFlame1.Enabled = false;
+		UDFlame2.Enabled = false;
+		IsFiring = false;
+		_currentJetFuel = MaxJetFuel;
+	}
+
+	protected override void FireFinished()
+	{
+		base.FireFinished();
+		if ( Equipment.Grub != null )
+			Equipment.Grub.Animator.IsOnJetpack = false;
+
+		FBFlame.Enabled = false;
+		UDFlame1.Enabled = false;
+		UDFlame2.Enabled = false;
+		IsFiring = false;
+		_currentJetFuel = MaxJetFuel;
+	}
+
+	[Property] public GameObject FBFlame { get; set; }
+
+	[Property] public GameObject UDFlame1 { get; set; }
+
+	[Property] public GameObject UDFlame2 { get; set; }
```

Comment:
I have no clue what these properties represent. Could we name them something a little more clear?

Reply:
Forward/Back flame
Up/Down flame